src/agents/workflows/subgraphs/step4_features/solution/pdf_utils.py

```
@@ -0,0 +1,318 @@
+    t = text.replace('\\', '\x01').replace('\x01\x01', '\x01').replace('\x01', '\\').strip()
+    if t.startswith('$') and t.endswith('$'): t = t[1:-1]
+    elif t.startswith('\\(') and t.endswith('\\)'): t = t[2:-2]
+    
+    t = re.sub(r'\\sum(?:_\{([^}]+)\})?(?:\^\{([^}]+)\})?', lambda m: f"Σ({m.group(1) or ''}→{m.group(2) or ''})", t)
+    t = re.sub(r'\\int(?:_\{([^}]+)\})?(?:\^\{([^}]+)\})?', lambda m: f"∫({m.group(1) or ''}→{m.group(2) or ''})", t)
+    t = re.sub(r'\\lim_\{([^}]+)\}', lambda m: "lim(" + m.group(1).replace('\\to', '→') + ")", t)
+    t = re.sub(r'\\(?:d|t)?frac\{([^}]+)\}\{([^}]+)\}', r'(\1/\2)', t)
+    t = re.sub(r'\\sqrt(?:\[([^\]]+)\])?\{([^}]+)\}', lambda m: f"{m.group(1) or ''}√{m.group(2)}", t)
+    
+    mapping = {
+        r'\alpha': 'α', r'\beta': 'β', r'\gamma': 'γ', r'\delta': 'δ', r'\epsilon': 'ε',
```

Ruff (0.14.14)

[warning] String contains ambiguous `α` (GREEK SMALL LETTER ALPHA). Did you mean `a` (LATIN SMALL LETTER A)?

(RUF001)